src/edu/washington/escience/myria/api/encoding/SamplingDistributionEncoding.java

```diff
@@ -0,0 +1,35 @@
+package edu.washington.escience.myria.api.encoding;
+
+import javax.ws.rs.core.Response;
+
+import edu.washington.escience.myria.api.MyriaApiException;
+import edu.washington.escience.myria.api.encoding.QueryConstruct.ConstructArgs;
+import edu.washington.escience.myria.operator.SamplingDistribution;
+
+public class SamplingDistributionEncoding extends UnaryOperatorEncoding<SamplingDistribution> {
+
+  /** A specific number of tuples to sample. */
+  public Integer sampleSize;
+
+  /** Percentage of total tuples to sample. */
+  public Float samplePercentage;
+
+  @Required
+  public String sampleType;
```

Comment:
See comment in SamplingDistribution.java, I think if you make this an enum, it should "just work" due to default Jackson magic.